python_modules/dagster/dagster/_utils/pydantic_yaml.py

```diff
@@ -32,25 +85,49 @@ def parse_yaml_file_to_pydantic(cls: Type[T], src: str, filename: str = "<string
             that the model corresponds to.
     """
     parsed = parse_yaml_with_source_positions(src, filename)
-    try:
-        model = parse_obj_as(cls, parsed.value)
-    except ValidationError as e:
-        if USING_PYDANTIC_1:
-            raise e
-        else:
-            line_errors = []
-            for error in e.errors():
-                key_path: KeyPath = error["loc"]
-                key_path_str = ".".join(str(part) for part in key_path)
-                source_position = parsed.source_position_tree.lookup(key_path)
-                line_errors.append({**error, "loc": [key_path_str + " at " + str(source_position)]})
+    return _parse_and_populate_model_with_annotated_errors(
+        cls=cls, obj_parse_root=parsed, file_root=parsed, obj_key_path_prefix=[]
+    )
 
-            raise ValidationError.from_exception_data(  # type: ignore
-                title=e.title,  # type: ignore
-                line_errors=line_errors,
-                input_type="json",
-                hide_input=False,
-            ) from None
 
-    populate_source_position_and_key_paths(model, parsed.source_position_tree)
-    return model
+def parse_yaml_file_to_pydantic_list(cls: Type[T], src: str, filename: str = "<string>") -> List[T]:
```

Comment:
```suggestion
def parse_yaml_file_to_pydantic_sequence(cls: Type[T], src: str, filename: str = "<string>") -> Sequence[T]:
```

Reply:
What is the motivation to keep this generic - is there a world in which we see ourselves returning a non-List sequence in the future?

Sequence input makes sense since it allows for more flexible input, but seems like the effect on output type is the inverse

Reply:
It ensures that callers don't modify it, which gives us more flexibility to change it in the future. More on this here: https://github.com/dagster-io/dagster/pull/21808#pullrequestreview-2059064883.

Reply:
The stakes are quite low here because it's an internal API, but I think it's a good habit to be in